src/cleaning/alru.c

```diff
@@ -59,6 +53,12 @@ struct alru_flush_ctx {
 	size_t flush_data_limit;
 };
 
+struct alru_context {
+	struct alru_flush_ctx flush_ctx;
+	env_spinlock list_lock;
```

Comment:
ALRU lists are maintained per partition, so there's no need for single global ALRU lock. Having this lock per partition would make it more fine grained.

Reply:
done